File: authorization-manager-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__GraphQLAuthorizationManager.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.ContextAttributes;
import se.curity.identityserver.sdk.attribute.SubjectAttributes;
import se.curity.identityserver.sdk.authorization.AuthorizationResult;
import se.curity.identityserver.sdk.authorization.GraphQLObligation;
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationActionAttributes;
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationManager;
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationResourceAttributes;

public final class ${pluginName}GraphQLAuthorizationManager implements GraphQLAuthorizationManager
{
    private final ${pluginName}AuthorizationManagerConfig _configuration;

    public ${pluginName}GraphQLAuthorizationManager(${pluginName}AuthorizationManagerConfig configuration)
    {
        _configuration = configuration;
    }

    @Override
    public AuthorizationResult<GraphQLObligation> getGraphQLAuthorizationResult(SubjectAttributes subjectAttributes,
                                                                                GraphQLAuthorizationActionAttributes graphQLAuthorizationActionAttributes,
                                                                                GraphQLAuthorizationResourceAttributes graphQLAuthorizationResourceAttributes,
                                                                                ContextAttributes contextAttributes)
    {
        // TODO: Perform decision and return appropriate AuthorizationResult
        return AuthorizationResult.notApplicable();
    }

}

File: kotlin-authorization-manager-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__GraphQLAuthorizationManager.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.attribute.ContextAttributes
import se.curity.identityserver.sdk.attribute.SubjectAttributes
import se.curity.identityserver.sdk.authorization.GraphQLObligation
import se.curity.identityserver.sdk.authorization.AuthorizationResult
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationActionAttributes
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationManager
import se.curity.identityserver.sdk.authorization.graphql.GraphQLAuthorizationResourceAttributes

class ${pluginName}GraphQLAuthorizationManager(private val _configuration: ${pluginName}AuthorizationManagerConfig): GraphQLAuthorizationManager
{
    override fun getGraphQLAuthorizationResult(subjectAttributes: SubjectAttributes,
                                               graphQLAuthorizationActionAttributes: GraphQLAuthorizationActionAttributes,
                                               graphQLAuthorizationResourceAttributes: GraphQLAuthorizationResourceAttributes,
                                               contextAttributes: ContextAttributes)
        : AuthorizationResult<GraphQLObligation>
    {
        // TODO: Perform decision and return appropriate AuthorizationResult
        TODO("Not implemented")
    }
}
